From 155cf5ff6f143c0d34d8d999fef238a4dd3cf55b Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 30 Nov 2022 09:51:27 +0200
Subject: [PATCH] [SW-66327] pif_arc: rename sts and err functions

Pif Arc will handle pcie init in addition to the FLR, and possibly
other functionality in the future. status and error reporting functions
are renamed to meet the wider functionality.

Change-Id: I4b7eaebb1a85ad5ffd3488690761af23686b4a3d
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/pcie_pif/pcie_pif.c b/app/pcie_pif/pcie_pif.c
index 400e13c..a05c925 100644
--- a/app/pcie_pif/pcie_pif.c
+++ b/app/pcie_pif/pcie_pif.c
@@ -44,7 +44,7 @@
 	},
 };
 
-void hl_flr_report_err(u32 err_bit)
+void hl_pif_report_err(u32 err_bit)
 {
 	u32 reg_val;
 
@@ -53,7 +53,7 @@
 	regmap_writel(reg_val, pif_arc.error_reg);
 }
 
-static void flr_report_status(u32 status_bit)
+static void hl_pif_report_sts(u32 status_bit)
 {
 	u32 reg_val;
 
@@ -62,7 +62,7 @@
 	regmap_writel(reg_val, pif_arc.status_reg);
 }
 
-static void flr_clear_status(u32 status_bit)
+static void hl_pif_clear_sts(u32 status_bit)
 {
 	u32 reg_val;
 
@@ -88,7 +88,7 @@
 	regmap_writel(1, pif_arc.pf_flr_done);
 	pif_ts_delay(FLR_ACTIVE_SIG_WIDTH_US);
 	regmap_writel(0, pif_arc.pf_flr_done);
-	flr_report_status(PIF_STAT_FLR_DONE_REPORTED);
+	hl_pif_report_sts(PIF_STAT_FLR_DONE_REPORTED);
 }
 
 static inline bool pif_flr_rst_en(void)
@@ -101,13 +101,13 @@
 	while (!regmap_readl(pif_arc.psoc2pif_die_ready)) {
 		/* meassuring timeout from FLR event start */
 		if (pif_ts_timeout_expired(0, FLR_SAFE_TIMEOUT_US)) {
-			hl_flr_report_err(PIF_ERR_SEC_DIE_READY_TIMEOUT);
+			hl_pif_report_err(PIF_ERR_SEC_DIE_READY_TIMEOUT);
 			break;
 		}
 	}
 
 	regmap_writel(0, pif_arc.psoc2pif_die_ready);
-	flr_report_status(PIF_STAT_SEC_DIE_READY);
+	hl_pif_report_sts(PIF_STAT_SEC_DIE_READY);
 }
 
 static void set_flr_active(void)
@@ -115,7 +115,7 @@
 	regmap_writel(1, pif_arc.pif2psoc_flr_active);
 	pif_ts_delay(FLR_ACTIVE_SIG_WIDTH_US);
 	regmap_writel(0, pif_arc.pif2psoc_flr_active);
-	flr_report_status(PIF_STAT_INT_RESET_ACTIVE);
+	hl_pif_report_sts(PIF_STAT_INT_RESET_ACTIVE);
 }
 
 static void poll_flr_done(void)
@@ -124,13 +124,13 @@
 	while (!regmap_readl(pif_arc.psoc2pif_flr_done)) {
 		/* meassuring timeout from FLR event start */
 		if (pif_ts_timeout_expired(0, FLR_SAFE_TIMEOUT_US)) {
-			hl_flr_report_err(PIF_ERR_FLR_HW_RESET_TIMEOUT);
+			hl_pif_report_err(PIF_ERR_FLR_HW_RESET_TIMEOUT);
 			break;
 		}
 	}
 
 	regmap_writel(0, pif_arc.psoc2pif_flr_done);
-	flr_clear_status(PIF_STAT_INT_RESET_ACTIVE);
+	hl_pif_clear_sts(PIF_STAT_INT_RESET_ACTIVE);
 }
 
 #define PIF_ISOLATE_OUTBOUND	BIT(0)
@@ -150,12 +150,12 @@
 	while (!regmap_readl(pif_arc.outstand_trans)) {
 		/* meassuring timeout from FLR event start */
 		if (pif_ts_timeout_expired(0, FLR_SAFE_TIMEOUT_US)) {
-			hl_flr_report_err(PIF_ERR_PCI_OUTBOUND_TIMEOUT);
+			hl_pif_report_err(PIF_ERR_PCI_OUTBOUND_TIMEOUT);
 			break;
 		}
 	}
 
-	flr_report_status(PIF_STAT_PCI_ACCESS_DISABLED);
+	hl_pif_report_sts(PIF_STAT_PCI_ACCESS_DISABLED);
 }
 
 static void enable_pci_access(void)
@@ -169,7 +169,7 @@
 	regval &= ~PIF_ISOLATE_OUTBOUND;
 	regmap_writel(regval, pif_arc.outbound_iso);
 
-	flr_clear_status(PIF_STAT_PCI_ACCESS_DISABLED);
+	hl_pif_clear_sts(PIF_STAT_PCI_ACCESS_DISABLED);
 }
 
 static void trigger_flr_reset(void)
@@ -190,8 +190,8 @@
 		       PIF_STAT_PCI_ACCESS_DISABLED | PIF_STAT_INT_RESET_ACTIVE |
 		       PIF_STAT_FLR_DONE_REPORTED;
 
-	flr_clear_status(clr_mask);
-	flr_report_status(PIF_STAT_FLR_EVENT_START);
+	hl_pif_clear_sts(clr_mask);
+	hl_pif_report_sts(PIF_STAT_FLR_EVENT_START);
 }
 
 void pci_dev_specific_run(void)
@@ -210,7 +210,7 @@
 	/* TODO: wait for confirmation from FW that security preparations have finished */
 
 	enable_pci_access();
-	flr_report_status(PIF_STAT_FLR_EVENT_FINISH);
+	hl_pif_report_sts(PIF_STAT_FLR_EVENT_FINISH);
 
 	pif_ts_rearm_enable(true);
 }
@@ -225,7 +225,7 @@
 	flr_sts.primary_die = hl_get_die_id() ? false : true;
 	flr_sts.flr_sw_ctrl = true;
 	flr_sts.setup_done = true;
-	flr_report_status(PIF_STAT_FLR_INITIALIZED);
+	hl_pif_report_sts(PIF_STAT_FLR_INITIALIZED);
 }
 
 void pci_dev_specific_disable(void)
@@ -294,7 +294,7 @@
 /* default exception handler, report error and halt */
 void exit_halt(void)
 {
-	hl_flr_report_err(PIF_ERR_CPU_EXCEPTION_HALT);
+	hl_pif_report_err(PIF_ERR_CPU_EXCEPTION_HALT);
 
 	wfe();
 }
diff --git a/include/preboot.h b/include/preboot.h
index aaa9ea8..c5f6344 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -260,7 +260,7 @@
 void preboot_dev_spec_legacy_vers_setup(enum preboot_vers vers_type, u32 offset);
 void ppboot_secure_arc(void);
 
-void hl_flr_report_err(u32 err_bit);
+void hl_pif_report_err(u32 err_bit);
 void preboot_security_sp_sram_init(void);
 u32 hl_get_pll_div_fix(void);
 
diff --git a/lib/common/pif_ts.c b/lib/common/pif_ts.c
index f2dbd28..80f2386 100644
--- a/lib/common/pif_ts.c
+++ b/lib/common/pif_ts.c
@@ -68,7 +68,7 @@
 				start = 0;
 				continue;
 			} else {
-				hl_flr_report_err(PIF_ERR_TS_COUNTER_OVERFLOW);
+				hl_pif_report_err(PIF_ERR_TS_COUNTER_OVERFLOW);
 				break;
 			}
 		}
@@ -87,7 +87,7 @@
 
 	current = pif_ts_get();
 	if (current == PIF_TIMESTAMP_MAX_VAL) {
-		hl_flr_report_err(PIF_ERR_TS_COUNTER_OVERFLOW);
+		hl_pif_report_err(PIF_ERR_TS_COUNTER_OVERFLOW);
 		return true;
 	}
 
